pkg/azurefile/nodeserver.go

```diff
@@ -329,6 +343,15 @@ func (d *Driver) NodeStageVolume(ctx context.Context, req *csi.NodeStageVolumeRe
 		}
 		klog.V(2).Infof("NodeStageVolume: volume %s format %s and mounting at %s successfully", volumeID, targetPath, diskPath)
 	}
+
+	if protocol == nfs || isDiskMount {
+		if volumeMountGroup != "" {
+			klog.V(2).Infof("set gid of volume(%s) as %s using fsGroupChangePolicy(%s)", volumeID, volumeMountGroup, fsGroupChangePolicy)
```

Comment:
@andyzhangx did you see my comment https://github.com/kubernetes-sigs/azurefile-csi-driver/issues/1012#issuecomment-1120196467 ? At least to me this does not seem right and we merged it.

Reply:
hi @gnufied, this is a workaround fix since nfs and smb are both supported with this driver. Is there any better way to fix this?

Reply:
There is no better way to fix this for VHD disks, but I am really concerned that if we take path taken in this code then customers will want to rely on it and when with a future version of CSI and k8s these changes are finalized - it will break their work-flow.

Reply:
thanks for raising the concern. currently `fsGroupChangePolicy` is an azure file volume property, I think we could do some backwards compatibility with future version of CSI and k8s these changes are finalized, we already hit some mount issues recently and need to workaround ASAP. thanks.

Name | Meaning | Example | Mandatory | Default value 
--- | --- | --- | --- | ---
fsGroupChangePolicy | indicates how volume's ownership will be changed by the driver, pod `securityContext.fsGroupChangePolicy` is ignored  | `OnRootMismatch`(by default), `Always`, `None` | No | `OnRootMismatch`